In [35]:
import pandas as pd
from sklearn.neural_network import MLPClassifier
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import confusion_matrix

In [36]:
df = pd.read_excel("wireless_mlpclassifier.xlsx")
data = df.iloc[:, 0:7]
data

[Low price (낮은 가격)]  [Noise cancelling (노이즈 캔슬링)]  \
0                     1                             4   
1                     3                             5   
2                     4                             3   
3                     3                             4   
4                     4                             5   
..                  ...                           ...   
81                    3                             4   
82                    6                             3   
83                    5                             6   
84                    4                             6   
85                    6                             2   

    [Battery time (배터리 시간)]  [Light weight (가벼운 무게)]  [Brand (브랜드)]  \
0                         3                        5              6   
1                         4                        6              2   
2                         5                        6              2   
3                         6                        5              1   
4                         2                        1              6   
..                      ...                      ...            ...   
81                        2                        5              6   
82                        1                        4              5   
83                        1                        2              4   
84                        3                        5              1   
85                        3                        5              1   

    [Sound Quality (음질)]  wish device  
0                      2            1  
1                      1            2  
2                      1            2  
3                      2            2  
4                      3            1  
..                   ...          ...  
81                     1            2  
82                     2            4  
83                     3            4  
84                     2            3  
85                     4            1  

[86 rows x 7 columns]

In [37]:
training_set, validation_set = train_test_split(data, test_size = 0.2, random_state = 21)

In [38]:
X_train = training_set.iloc[:,0:6].values
y_train = training_set.iloc[:, -1].values
X_val = validation_set.iloc[:,0:6].values
y_val = validation_set.iloc[:, -1].values

In [39]:
def accuracy(confusion_matrix):
    diagonal_sum = confusion_matrix.trace()
    sum_of_all_elements = confusion_matrix.sum()
    return diagonal_sum/sum_of_all_elements

In [40]:
classifier = MLPClassifier(max_iter= 300,solver = 'adam', random_state = 1)
'''hidden_layer_sizes = (150,100,50),\
                           
    >> 이거 없애면 accuracy = 0.2777777777777778'''

'hidden_layer_sizes = (150,100,50),                           \n    >> 이거 없애면 accuracy = 0.2777777777777778'

In [41]:
classifier.fit(X_train, y_train)

C:\Users\jiwon\anaconda3\lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:614: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (300) reached and the optimization hasn't converged yet.
  warnings.warn(


MLPClassifier(max_iter=300, random_state=1)

In [42]:
y_pred = classifier.predict(X_val)

In [43]:
cm = confusion_matrix(y_pred, y_val)

In [44]:
accuracy(cm)

0.3333333333333333

scaler 없애면 accuracy 0.2777777777777778

hidden layer도 디폴트면 0.3333333333333333

다른것들은 디폴트로 돌려도 정확도 변화 없음.